src/konnektor/network_planners/generators/maximal_network_generator.py

```diff
@@ -14,35 +14,36 @@ class MaximalNetworkGenerator(NetworkGenerator):
     def __init__(
         self,
         mappers: AtomMapper | list[AtomMapper],
-        scorer,
+        scorer: Callable,
         progress: bool = False,
         n_processes: int = 1,
     ):
         """
-        The `MaximalNetworkGenerator` builds a fully connected graph for given set of `Component` s.
-        It assumes each `Component` can be connected to every other `Component`.
-        The `Transformation` s of this graph are realized as `AtomMapping` s of pairwise `Component` s.
+        The ``MaximalNetworkGenerator`` attempts to build a fully connected graph (every node connected to every other node) for given set of `Component`/s.
+
+        The edges of the graph are ``Transformation`` s, which contain `AtomMapping` s of pairwise `Component`/s.
         If not all mappings can be created, it will ignore the mapping failure and return a nearly fully connected graph.
 
+        If multiple ``AtomMapper``/s are provided, but no scorer, the *first valid* ``AtomMapper`` provided will be used.
+
         ... note::
         This approach is not recommended for Free Energy calculations in application cases, as it is very computationally expensive.
         However, this approach is very important, as all other approaches use the Maximal Network as an initial solution,
         then remove edges to achieve the desired design.
 
-        This class is recommended as an `initial_edge_lister` for other approaches.
-        The `MaximalNetworkGenerator` is parallelized and the number of CPUs can be given with `n_processes`.
-        All other approaches in Konnektor benefit from this parallelization and you can use this parallelization with `n_processes` key word during class construction.
+        This class is recommended as an ``initial_edge_lister`` for other network generators.
+        The ``MaximalNetworkGenerator`` is parallelized and the number of CPUs can be chosen with the ``n_processes`` argument.
 
         Parameters
         ----------
         mappers: Union[AtomMapper, list[AtomMapper]]
-            the atom mapper is required, to define the connection between two ligands.
-        scorer: AtomMappingScorer
-            scoring function evaluating an atom mapping, and giving a score between [0,1].
+            ``AtomMapper`` to use to define the relationship between two ligands.
+        scorer: Callable
```

Comment:
Docstring above says you can pass no scorer, but here & the type hint it's not Optional. Briefly looking at the code I have no idea if it should or shouldn't.

Reply:
scorer is optional, this was me being sloppy with type hints! there's a code block for if a scorer is not passed. thanks for catching this!

Reply:
updated here as well now!